distributed/tests/test_client.py

```diff
@@ -5652,23 +5651,17 @@ async def test_config_scheduler_address(s, a, b):
 
 @gen_cluster(client=True)
 async def test_warn_when_submitting_large_values(c, s, a, b):
-    with warnings.catch_warnings(record=True) as record:
+    with pytest.warns(
+        UserWarning,
+        match=r"Large object of size (2\.00 MB|1.91 MiB) detected in task graph:\s+"
+        r"\(b'0+ \.\.\. 0+',\)\s+"
+        r"Consider scattering large objects ahead of time.*",
+    ) as exc_info:
         future = c.submit(lambda x: x + 1, b"0" * 2000000)
 
-    text = str(record[0].message)
-    assert "2.00 MB" in text or "1.91 MiB" in text
-    assert "large" in text
-    assert "..." in text
-    assert "'000" in text
-    assert "000'" in text
-    assert len(text) < 2000
-
-    with warnings.catch_warnings(record=True) as record:
-        data = b"0" * 2000000
-        for i in range(10):
-            future = c.submit(lambda x, y: x, data, i)
-
-    assert len(record) < 2
```

Comment:
IIUC it looks like we're dropping this check

Reply:
I've now restored this as `assert not record`